src/client/pages/Debugger/components/Offer.tsx

```diff
@@ -24,63 +38,119 @@ enum QuoteType {
+    },
+  ],
+  SE: [
+    {
+      label: 'Swedish Apartment',
+      value: QuoteType.SwedishApartment,
+      initialFormValues: initialSeApartmentValues,
+    },
+    {
+      label: 'Swedish House',
+      value: QuoteType.SwedishHouse,
+    },
+  ],
+}
+
+const getCurrentAvailableQuoteData = (
+  currentMarket: Market,
+  currentQuoteType: QuoteType,
+) => {
+  const marketQuoteTypes = quotesByMarket[currentMarket]
+  const currentQuoteTypeData = marketQuoteTypes.find(
+    ({ value }) => value === currentQuoteType,
+  )
+  return currentQuoteTypeData
+}
+
+export const Offer: React.FC<OfferProps> = ({ sessionToken }) => {
   const [quoteId, setQuoteId] = useState<string>('') // TODO handle multiple quotes
+  const [getQuote, { data, refetch }] = useQuoteLazyQuery()
+  const [hasQuoteCreatingError, setHasQuoteCreatingError] = useState(false)
   const storageState = useStorage()
   const currentLocale = useCurrentLocale()
   const localeIsoCode = getLocaleIsoCode(currentLocale)
   const currentMarket = useMarket()
 
-  const quotesByMarket = {
-    DK: [
-      {
-        label: 'Danish Home',
-        value: QuoteType.DanishHome,
-      },
-    ],
-    NO: [
-      {
-        label: 'Norwegian Home',
-        value: QuoteType.NorwegianHome,
-      },
-      {
-        label: 'Norwegian Travel',
-        value: QuoteType.NorwegianTravel,
-      },
-    ],
-    SE: [
-      {
-        label: 'Swedish Apartment',
-        value: QuoteType.SwedishApartment,
-      },
-      {
-        label: 'Swedish House',
-        value: QuoteType.SwedishHouse,
-      },
-    ],
-  }
-
   const [quoteType, setQuoteType] = useState(
     quotesByMarket[currentMarket][0].value,
   )
 
+  const handleSubmit = async (
+    values: Values,
+    storage: Record<string, unknown>,
+  ) => {
+    if (sessionToken) {
+      await createQuote(
+        storage,
+        localeIsoCode,
+      )({
+        input: {
+          ...values,
+          id: quoteId,
+          currentInsurer: values.currentInsurer || undefined,
+          // @ts-ignore
+          startDate: values.startDate || undefined,
+        },
+      }).catch(() => setHasQuoteCreatingError(true))
+
+      if (refetch) {
+        await refetch()
+      }
+    }
+  }
+
   useEffect(() => {
     const quoteIds = storageState.session.getSession()?.quoteIds ?? []
+
     if (!quoteId && quoteIds[0]) {
       setQuoteId(quoteIds[0] ?? '')
       return
     }
 
-    if (quoteId) {
-      storageState.session.setSession({
-        ...storageState.session.getSession(),
-        quoteIds: [quoteId],
-      })
-      getQuote({ variables: { id: quoteId, perilsLocale: localeIsoCode } })
+    if (!quoteId && !quoteIds.length) {
+      setQuoteId(uuid())
     }
+
+    storageState.session.setSession({
+      ...storageState.session.getSession(),
+      quoteIds: [quoteId],
+    })
   }, [quoteId])
 
+  useEffect(() => {
+    if (quoteId?.length === 36 && sessionToken) {
```

Comment:
If we disabled the uuid input field you can remove the length check

Reply:
OK, since we agreed on keeping the UUID input field as it is (for now, maybe adding a button to generate a UUID in the future) I will resolve this conversation! 😇